google/cloud/internal/random.cc

```diff
@@ -13,11 +13,51 @@
 // limitations under the License.
 
 #include "google/cloud/internal/random.h"
+#include <algorithm>
+#include <limits>
 
 namespace google {
 namespace cloud {
 inline namespace GOOGLE_CLOUD_CPP_NS {
 namespace internal {
+std::vector<unsigned int> FetchEntropy(std::size_t desired_bits) {
+  // We use the default C++ random device to generate entropy.  The quality of
+  // this source of entropy is implementation-defined. The version in
+  // Linux with libstdc++ (the most common library on Linux) is based on
+  // either `/dev/urandom`, or (if available) the RDRAND, or the RDSEED CPU
+  // instruction.
+  //     http://en.cppreference.com/w/cpp/numeric/random/random_device/random_device
+  //
+  // On Linux with libc++ it is also based on `/dev/urandom`, but it is not
+  // known if it uses the RDRAND CPU instruction (though `/dev/urandom` does).
```

Comment:
I'm not sure what this means.  libc++ uses /dev/urandom, and /dev/urandom uses RDRAND, but we don't know if libc++ uses RDRAND?

Reply:
Reworded the whole comment, I think it is clearer now. PTAL.